gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -752,6 +756,27 @@ case class ColumnarOverrideRules(session: SparkSession)
     fallbacks >= wholeStageFallbackThreshold
   }
 
+  def fallbackWholeQuery(plan: SparkPlan): Boolean = {
+    var fallbacks = 0
+    def countFallback(plan: SparkPlan): Unit = {
+      plan match {
+        // Another stage.
+        case _: QueryStageExec =>
+          return
+        case ColumnarToRowExec(p: GlutenPlan) =>
+          logDebug(s"cr2: ${p}")
+          fallbacks = fallbacks + 1
+        // Possible fallback for leaf node.
+        case leafPlan: LeafExecNode if !leafPlan.isInstanceOf[GlutenPlan] =>
+          fallbacks = fallbacks + 1
+        case _ =>
+      }
+      plan.children.map(p => countFallback(p))
+    }
+    countFallback(plan)
+    fallbacks > 1
```

Comment:
can we consider to use `wholeStageFallbackThreshold` as threshold too for query policy ?

Reply:
yes, let me try to add another configurable value for this